### PR TITLE
#57745: Cannot save unchanged but auth-restricted meta value

### PR DESCRIPTION
**Pull request by kadamwhite**
Trac ticket: https://core.trac.wordpress.org/ticket/57745

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/rest-api/rest-post-meta-fields.php [hunk 1]
```diff
@@ -2301,6 +2301,45 @@ public function test_update_meta_with_unchanged_object_values() {
...
+			'authenticated',
+			array(
+				'single'        => true,
+				'type'          => 'boolean',
+				'default'       => false,
+				'show_in_rest'  => true,
+				'auth_callback' => static function() {
+					// No user has permissions to update this field.
+					return false;
+				},
```
Comment (spacedmonkey) [inline tests/phpunit/tests/rest-api/rest-post-meta-fields.php]:
```suggestion
				'auth_callback' => '__return_false'
```

**Review by spacedmonkey — APPROVED** [review]
One nitpic, but this looks good to commit to me.

File: tests/phpunit/tests/rest-api/rest-post-meta-fields.php [hunk 2]
```diff
@@ -2301,6 +2301,42 @@ public function test_update_meta_with_unchanged_object_values() {
...
+	public function test_update_meta_with_unchanged_values_and_custom_authentication() {
+		register_post_meta(
+			'post',
+			'authenticated',
+			array(
+				'single'        => true,
+				'type'          => 'boolean',
+				'default'       => false,
+				'show_in_rest'  => true,
+				'auth_callback' => '__return_false'
```
Comment (kadamwhite) [inline tests/phpunit/tests/rest-api/rest-post-meta-fields.php]:
```suggestion
				'auth_callback' => '__return_false',
```

**Comment by kadamwhite** [comment]
Merged in https://core.trac.wordpress.org/changeset/56075 / 507cab8d6f5d469c6b0131bb07a45eb6f0c6724e

**Pull request closed without merging**